x-pack/platform/plugins/shared/lens/public/react_embeddable/renderer/lens_embeddable_component.tsx

```diff
@@ -43,7 +43,10 @@ export function LensEmbeddableComponent({
     api.rendered$,
     api.viewMode$
   );
-  const canEdit = Boolean(api.isEditingEnabled?.() && getViewMode(latestViewMode) === 'edit');
+  const canEdit = Boolean(
+    api.isEditingEnabled?.() &&
+      [latestViewMode, getViewMode(api.viewMode$)].some((mode) => mode === 'edit')
```

Comment:
I think this should be....

```suggestion
      [latestViewMode, getViewMode(api)].some((mode) => mode === 'edit')
```

The `getViewMode` expects the `api` directly and checks for `viewMode$`. These `unknown` arg types are dangerous!! 💥

Reply:
Thanks @nickofthyme , well spotted.